projects/packages/forms/src/dashboard/inbox/export-responses/google-drive.js

```diff
@@ -130,14 +149,15 @@ const GoogleDriveExport = ( { onExport } ) => {
 						</Button>
 					) }
 
-					{ ! isConnected && isUserConnected && (
+					{ ! isConnectedToGoogleDrive && isUserConnected && (
 						<Button
 							href={ config( 'gdriveConnectURL' ) }
 							className={ buttonClasses }
 							variant="primary"
 							rel="noopener noreferrer"
 							target="_blank"
 							onClick={ handleConnectClick }
+							ref={ connectButtonRef }
```

Comment:
Why not simply use [Callback Refs](https://legacy.reactjs.org/docs/refs-and-the-dom.html#callback-refs)?

```suggestion
							ref={ el => autoConnect && el?.click() }
```

That way you can get rid of `useEffect`, the timeout and the state management for `hasAutoConnected`

Reply:
Beautiful simplification!